pkgs/applications/audio/projectm/default.nix

```diff
@@ -1,57 +1,66 @@
-{ stdenv, fetchurl, fetchpatch, pkgconfig, cmake
-, glew, ftgl, ttf_bitstream_vera
-, withQt ? true, qt4
-, withLibvisual ? false, libvisual, SDL
-, withJack ? false, libjack2
-, withPulseAudio ? true, libpulseaudio
-}:
+{ mkDerivation
+, lib
+, fetchFromGitHub
 
-assert withJack       -> withQt;
-assert withPulseAudio -> withQt;
+, autoreconfHook
+, pkgconfig
+, SDL2
+, qtdeclarative
+, libpulseaudio
+, glm
+, which
+}:
 
-stdenv.mkDerivation {
-  name = "projectm-2.1.0";
+mkDerivation rec {
+  pname = "projectm";
+  version = "3.1.3";
 
-  meta = {
-    description = "Music Visualizer";
-    homepage = http://projectm.sourceforge.net/;
-    license = stdenv.lib.licenses.lgpl21Plus;
-    platforms = stdenv.lib.platforms.linux;
+  src = fetchFromGitHub {
+    owner = "projectM-visualizer";
+    repo = "projectM";
+    rev = "v${version}";
+    sha256 = "1mjnahr694phksmvc069y89rv85s4l2z9fixkc3l1f5qj2vgn4sy";
   };
 
-  src = fetchurl {
-    url = "mirror://sourceforge/projectm/2.1.0/projectM-complete-2.1.0-Source.tar.gz";
-    sha256 = "1vh6jk68a0jdb6qwppb6f8cbgmhnv2ba3bcavzfd6sq06gq08cji";
-  };
+  nativeBuildInputs = [
+    pkgconfig
+    autoreconfHook
+    which
+  ];
 
-  patch_gcc6 = fetchpatch {
-    url = https://raw.githubusercontent.com/gentoo/gentoo/45abd63abc6644b6e177c057b5b42d894dbf8e29/media-libs/libprojectm/files/libprojectm-2.1.0-fix-c++14.patch;
-    sha256 = "1i50scxv1jlqvb3jm3sql89a7wqckxhlpvnhz20vvmm1kii6lrsn";
-  };
+  outputs = [ "bin" "dev" "out" ]; # NOTE: 2019-10-05: libprojectM in "lib" creates a cyclic dependency on the "out"
 
-  patchPhase = ''
-    patch -d src/libprojectM -p1 -i "$patch_gcc6"
-    sed -i 's:''${LIBVISUAL_PLUGINSDIR}:''${CMAKE_INSTALL_PREFIX}/lib/libvisual-0.4:' \
-      src/projectM-libvisual/CMakeLists.txt
-  '';
-
-  nativeBuildInputs = [ pkgconfig cmake ];
+  buildInputs = [
+    SDL2
+    qtdeclarative
+    libpulseaudio
+    glm
+  ];
 
-  cmakeFlags = [
-    "-DprojectM_FONT_MENU=${ttf_bitstream_vera}/share/fonts/truetype/VeraMono.ttf"
-    "-DprojectM_FONT_TITLE=${ttf_bitstream_vera}/share/fonts/truetype/Vera.ttf"
-    "-DINCLUDE-PROJECTM-TEST=OFF"
-    "-DINCLUDE-PROJECTM-QT=${if withQt then "ON" else "OFF"}"
-    "-DINCLUDE-PROJECTM-LIBVISUAL=${if withLibvisual then "ON" else "OFF"}"
-    "-DINCLUDE-PROJECTM-JACK=${if withJack then "ON" else "OFF"}"
-    "-DINCLUDE-PROJECTM-PULSEAUDIO=${if withPulseAudio then "ON" else "OFF"}"
+  configureFlags = [
+    "--enable-qt"
+    "--enable-sdl"
   ];
 
-  buildInputs = with stdenv.lib;
-    [ glew ftgl ]
-    ++ optional withQt qt4
-    ++ optionals withLibvisual [ libvisual SDL ]
-    ++ optional withJack libjack2
-    ++ optional withPulseAudio libpulseaudio
-    ;
+  fixupPhase = ''
+    # NOTE: 2019-10-05: Upstream inserts the src path buring build into ELF rpath, so must delete it out
+    # upstream report: https://github.com/projectM-visualizer/projectm/issues/245
+    for entry in $out/bin/* ; do
```

Comment:
How is this supposed to work if this is not `$bin/bin/*`?

Reply:
Good question. @Anton-Latukha came up with this in #70433 and I missed it.